lib/plugins/aws/package/compile/events/eventBridge/index.js

```diff
@@ -108,119 +112,336 @@ class AwsCompileEventBridgeEvents {
+              'event-bus/*',
+            ],
           ],
-        ],
+        },
+        Action: ['events:CreateEventBus', 'events:DeleteEventBus'],
+      });
+      hasEventBusesIamRoleStatement = true;
+    }
+    return {
+      iamRoleStatements,
+      hasEventBusesIamRoleStatement,
+    };
+  }
+
+  _compileWithCloudFormation({
+    eventBusName: _eventBusName,
+    EventBus,
+    compiledCloudFormationTemplate,
+    functionName,
+    RuleName,
+    Input,
+    InputPath,
+    InputTransformer,
+    Pattern,
+    Schedule,
+    FunctionName,
+    idx,
+  }) {
+    let eventBusResource;
+    let eventBusExists = false;
+    let eventBusName = _eventBusName;
+
+    // It suggests that the object already exists and is being imported
+    if (_.isObject(eventBusName)) {
+      eventBusExists = true;
+    }
+
+    // Does the resource already exist? ARN string - assume it is valid - CF will validate ultimately
+    if (typeof eventBusName === 'string' && eventBusName.startsWith('arn')) {
+      eventBusExists = true;
+      eventBusName = EventBus.slice(EventBus.indexOf('/') + 1);
+    }
+
+    const shouldCreateEventBus = !eventBusExists && eventBusName && eventBusName !== 'default';
+    if (shouldCreateEventBus) {
+      // Create EventBus Resource
+      eventBusResource = {
+        Type: 'AWS::Events::EventBus',
+        Properties: {
+          Name: eventBusName,
+        },
       };
-      iamRoleStatements.push({
-        Effect: 'Allow',
-        Resource: functionResources,
-        Action: ['lambda:AddPermission', 'lambda:RemovePermission'],
+
+      _.merge(compiledCloudFormationTemplate.Resources, {
+        [this.provider.naming.getEventBridgeEventBusLogicalId(eventBusName)]: eventBusResource,
       });
     }
 
+    const targetBase = {
+      Arn: {
+        'Fn::GetAtt': [this.provider.naming.getLambdaLogicalId(functionName), 'Arn'],
+      },
+      Id: makeEventBusTargetId(RuleName),
+    };
+
+    const target = this.addInputConfigToTarget({
+      target: targetBase,
+      Input,
+      InputPath,
+      InputTransformer,
+    });
+
+    // Create a rule
+    const eventRuleResource = {
+      Type: 'AWS::Events::Rule',
+      Properties: {
+        // default event bus is used when EventBusName is not set
+        EventBusName: eventBusName === 'default' ? undefined : eventBusName,
+        EventPattern: JSON.stringify(Pattern),
+        Name: RuleName,
+        ScheduleExpression: Schedule,
+        State: 'ENABLED',
+        Targets: [target],
+      },
+    };
+    // If this stack is creating the event bus the rule must depend on it to ensure stack can be removed
+    if (shouldCreateEventBus) {
+      eventRuleResource.DependsOn = this.provider.naming.getEventBridgeEventBusLogicalId(
+        eventBusName
+      );
+    }
+
+    const ruleNameLogicalIdStub = makeRuleName({
+      functionName: FunctionName,
+      index: idx,
+    });
+    _.merge(compiledCloudFormationTemplate.Resources, {
```

Comment:
Let's simply assign as:

```javascript
compiledCloudFormationTemplate.Resources[this.provider.naming.getEventBridgeRuleLogicalId(ruleNameLogicalIdStub)] = eventRuleResource;
```

(we have highly overused `_.merge` in our codebase which is dedicated for deep object merges)